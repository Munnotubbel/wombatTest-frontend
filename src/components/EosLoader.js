import React from "react";

const EosLoader = () => {
  const eosIcon =
    "https://res.cloudinary.com/munnotubbel/image/upload/v1581182752/wombat/5a52232c2f93c7a8d5137fdd_izhtxc.png";
  return <img alt="eosLoader" className="pulse" src={eosIcon} />;
};

export default EosLoader;
