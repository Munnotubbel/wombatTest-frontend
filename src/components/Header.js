import React, { useContext } from "react";
import { DataContext } from "../context/DataContext";
const Header = () => {
  const dataCon = useContext(DataContext);
  const isLoading = dataCon ? dataCon.isLoading : true;
  const accountName = dataCon ? dataCon.accountName : "nobody";

  return (
    <div className="headtitle">
      {isLoading === true ? (
        <h1>loading</h1>
      ) : (
        <h1>{accountName !== null ? accountName : "nobody"}</h1>
      )}
    </div>
  );
};

export default Header;
