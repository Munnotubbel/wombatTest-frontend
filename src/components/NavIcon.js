import React from "react";
import PropTypes from "prop-types";

const NavIcon = props => {
  const iconurl = props ? props.iconurl : "";
  const routename = props ? props.routename : "";

  return (
    <div className="navicon">
      <img alt="NavButton" src={iconurl} />
      <p>{routename}</p>
    </div>
  );
};

NavIcon.propTypes = {
  routename: PropTypes.string,
  iconurl: PropTypes.string
};

export default NavIcon;
