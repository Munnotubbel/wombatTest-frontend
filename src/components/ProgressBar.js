import React, { useEffect, useContext } from "react";
import PropTypes from "prop-types";
import { DataContext } from "../context/DataContext";

const ProgressBar = props => {
  const dataCon = useContext(DataContext);
  const loading = dataCon ? dataCon.isLoading : true;
  const percent = props ? props.percent : 0;
  const slug = props ? props.slug : Math.random.toString;
  const bool = loading === true;
  useEffect(() => move(), [bool]);

  var i = 0;
  function move() {
    var elem = document.getElementById(slug);
    if (percent > 0) {
      if (i === 0) {
        i = 1;

        var width = 0;
        var id = setInterval(frame, 20);
        function frame() {
          if (width >= percent) {
            clearInterval(id);
            i = 0;
          } else {
            width++;
            elem.style.width = width + "%";
          }
        }
      }
    } else {
      elem.style.width = percent + "%";
    }
  }

  return (
    <div className="progressbar">
      <div className="percent">
        <div>{isNaN(percent) ? "-" : percent} %</div>
      </div>
      <div className="barOuter">
        <div
          id={slug}
          style={{
            width: "0%",
            maxWidth: "100%",
            backgroundColor: "white",
            borderColor: "black",
            borderWidth: "2px",
            height: "16px",
            borderStyle: "solid"
          }}
        ></div>
      </div>
    </div>
  );
};

ProgressBar.propTypes = {
  slug: PropTypes.string,
  percent: PropTypes.number
};

export default ProgressBar;
