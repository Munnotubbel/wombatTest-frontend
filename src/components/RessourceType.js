import React from 'react'
import PropTypes from 'prop-types';

const RessourceType = (props) => {
    const eosIcon="https://res.cloudinary.com/munnotubbel/image/upload/v1581182752/wombat/5a52232c2f93c7a8d5137fdd_izhtxc.png"
    const type = props ? props.type : " - ";
    const consumption = props ? props.consumption : "";  
    const staked = props ? props.staked : (0).toFixed(4);
    const withIcon = props ? props.withIcon : false;

    return (
        <div className="ressource_type">
            <div id="type">{type}</div>
            <div id="consumption">{consumption}</div>
          
            <div className="stakedinfo"><p>{staked}</p>{withIcon===true ?<img className="coinIcon" alt="EOS-ICON" src={eosIcon}/>:null}</div>
        </div>
    )
}

RessourceType.propTypes = {
    type: PropTypes.string,
    consumption: PropTypes.string,
    staked: PropTypes.string,
    withIcon: PropTypes.bool
    
  };

export default RessourceType;
