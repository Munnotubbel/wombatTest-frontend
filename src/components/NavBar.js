import React from "react";
import NavIcon from "./NavIcon";
import { NavLink } from "react-router-dom";

const NavBar = () => {
  const buttonImgUrls = [
    "https://res.cloudinary.com/munnotubbel/image/upload/v1581240405/wombat/gear_fnbjbo.png",
    "https://res.cloudinary.com/munnotubbel/image/upload/v1581240405/wombat/wallet_qg6uwn.png",
    "https://res.cloudinary.com/munnotubbel/image/upload/v1581240455/wombat/home_shuqs8.png",
    "https://res.cloudinary.com/munnotubbel/image/upload/v1581240405/wombat/compass_wzpihn.png"
  ];

  return (
    <div className="navbar">
      <NavLink to="/settings" activeClassName="activeRoute">
        <NavIcon iconurl={buttonImgUrls[0]} routename={"Settings"} />
      </NavLink>
      <NavLink to="/wallet" activeClassName="activeRoute">
        <NavIcon iconurl={buttonImgUrls[1]} routename={"Wallet"} />
      </NavLink>
      <NavLink to="/home" activeClassName="activeRoute">
        <NavIcon iconurl={buttonImgUrls[2]} routename={"Home"} />
      </NavLink>
      <NavLink to="/explorer" activeClassName="activeRoute">
        <NavIcon iconurl={buttonImgUrls[3]} routename={"Explorer"} />
      </NavLink>
    </div>
  );
};

export default NavBar;
