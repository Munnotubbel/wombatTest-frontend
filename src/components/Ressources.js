import React, { useContext } from "react";
import RessourceType from "./RessourceType";
import ProgressBar from "./ProgressBar";
import { DataContext } from "../context/DataContext";

const Ressources = () => {
  const dataCon = useContext(DataContext);
  const net_use = dataCon ? dataCon.net_use : 0;
  const net_max = dataCon ? dataCon.net_max : 0;
  const net_weight = dataCon ? dataCon.net_weight : (0).toFixed(4);
  const cpu_use = dataCon ? dataCon.cpu_use : 0;
  const cpu_max = dataCon ? dataCon.cpu_max : 0;
  const cpu_weight = dataCon ? dataCon.cpu_weight : (0).toFixed(4);
  const ram_use = dataCon ? dataCon.ram_use : 0;
  const ram_max = dataCon ? dataCon.ram_max : 0;

  function calcPercent(number1, number2) {
    const percent = Math.round((number1 / number2) * 100);

    if (percent >= 100) {
      return 100;
    } else if (percent >= 0) {
      return percent;
    } else {
      return 0;
    }
  }

  return (
    <div className="ressources">
      <div>
        <RessourceType
          type={"Net"}
          consumption={`${net_use}kb / ${net_max}kb`}
          staked={net_weight}
          withIcon={true}
        />
        <ProgressBar percent={calcPercent(net_use, net_max)} slug={"netBar"} />
      </div>

      <div>
        <RessourceType
          type={"Cpu"}
          consumption={`${cpu_use}kb / ${cpu_max}kb`}
          staked={cpu_weight}
          withIcon={true}
        />
        <ProgressBar percent={calcPercent(cpu_use, cpu_max)} slug={"cpuBar"} />
      </div>

      <div style={{ marginBottom: "60px" }}>
        <RessourceType
          type={"Ram"}
          consumption={`${ram_use}kb / ${ram_max}kb`}
          staked={""}
        />
        <ProgressBar percent={calcPercent(ram_use, ram_max)} slug={"ramBar"} />
      </div>
    </div>
  );
};

export default Ressources;
