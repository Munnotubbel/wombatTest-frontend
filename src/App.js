import React, { useState } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./css/App.css";

import NavBar from "./components/NavBar";
import DataContextProvider from "./context/DataContext";
import Wallet from "./routes/Wallet";
import Settings from "./routes/Settings";
import Explorer from "./routes/Explorer";
import Home from "./routes/Home";
import Header from "./components/Header";

function App() {

  const screenModeMargin = window.matchMedia("(orientation: portrait)").matches
    ? "0%"
    : "4%";

  const [browserHeight, setBrowserHeight] = useState(window.innerHeight);

  window.addEventListener("orientationchange", function() {
    var afterOrientationChange = function() {
      setBrowserHeight(window.innerHeight);
      window.removeEventListener("resize", afterOrientationChange);
    };
    window.addEventListener("resize", afterOrientationChange);
  });

  return (
    <Router>
      <DataContextProvider>
        <div style={{ height: "100%" }}>
          <div
            className="outerContainer"
            style={{ height: browserHeight, marginTop: screenModeMargin }}
          >
            <div className="innerContainer">
              <div className="header">
                <Header />
              </div>

              <div className="info">
                <Switch>
                  <Route path="/settings">
                    <Settings />
                  </Route>
                  <Route path="/home">
                    <Home />
                  </Route>
                  <Route path="/explorer">
                    <Explorer />
                  </Route>
                  <Route exact path="/wallet">
                    <Wallet />
                  </Route>
                </Switch>
              </div>

              <div className="footer">
                <NavBar />
              </div>
            </div>
          </div>
        </div>
      </DataContextProvider>
    </Router>
  );
}

export default App;
