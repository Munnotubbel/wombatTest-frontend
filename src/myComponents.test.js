import React from 'react';
import { mount, shallow, render } from "enzyme";
import App from './App';
import EosLoader from "./components/EosLoader"
import Header from "./components/Header"
import NavBar from "./components/NavBar"
import NavIcon from "./components/NavIcon"
import ProgressBar from "./components/ProgressBar"
import Ressources from "./components/Ressources"
import RessourceType from "./components/RessourceType"
import TradeButtons from "./components/TradeButtons"

import Explorer from "./routes/Explorer"
import Home from "./routes/Home"
import Settings from "./routes/Settings"
import Wallet from "./routes/Wallet"


describe('Full App', () => {
  it('should render correctly in', () => {
    const component = shallow(<App />);
  
    expect(component).toMatchSnapshot();
  });
});


describe('test EosLoader', () => {
  it('should render correctly in', () => {
    const component = shallow(<EosLoader />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test Header', () => {
  it('should render correctly in', () => {
    const component = shallow(<Header />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test NavBar', () => {
  it('should render correctly in', () => {
    const component = shallow(<NavBar />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test NavIcon', () => {
  it('should render correctly in', () => {
    const component = shallow(<NavIcon />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test ProgressBar', () => {
  it('should render correctly in', () => {
    const component = shallow(<ProgressBar />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test Ressources', () => {
  it('should render correctly in', () => {
    const component = shallow(<Ressources />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test RessourceType', () => {
  it('should render correctly in', () => {
    const component = shallow(<RessourceType />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test TradeButtons', () => {
  it('should render correctly in', () => {
    const component = shallow(<TradeButtons />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test Explorer', () => {
  it('should render correctly in', () => {
    const component = shallow(<Explorer />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test Home', () => {
  it('should render correctly in', () => {
    const component = shallow(<Home />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test Settings', () => {
  it('should render correctly in', () => {
    const component = shallow(<Settings />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('test Wallet', () => {
  it('should render correctly in', () => {
    const component = shallow(<Wallet />);
  
    expect(component).toMatchSnapshot();
  });
});

