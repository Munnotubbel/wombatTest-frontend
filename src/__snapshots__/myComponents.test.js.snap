// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full App should render correctly in 1`] = `
<BrowserRouter>
  <DataContextProvider>
    <div
      className="outerContainer"
    >
      <div
        className="innerContainer"
      >
        <div
          className="header"
        >
          <Header />
        </div>
        <div
          className="info"
        >
          <Switch>
            <Route
              path="/settings"
            >
              <Settings />
            </Route>
            <Route
              path="/home"
            >
              <Home />
            </Route>
            <Route
              path="/explorer"
            >
              <Explorer />
            </Route>
            <Route
              exact={true}
              path="/wallet"
            >
              <Wallet />
            </Route>
          </Switch>
        </div>
        <div
          className="footer"
        >
          <NavBar />
        </div>
      </div>
    </div>
  </DataContextProvider>
</BrowserRouter>
`;

exports[`test EosLoader should render correctly in 1`] = `
<img
  alt="eosLoader"
  className="pulse"
  src="https://res.cloudinary.com/munnotubbel/image/upload/v1581182752/wombat/5a52232c2f93c7a8d5137fdd_izhtxc.png"
/>
`;

exports[`test Explorer should render correctly in 1`] = `
<div
  className="tobeadded"
>
  <h1>
    Explorer
  </h1>
</div>
`;

exports[`test Header should render correctly in 1`] = `
<div
  className="headtitle"
>
  <h1>
    loading
  </h1>
</div>
`;

exports[`test Home should render correctly in 1`] = `
<div
  className="tobeadded"
>
  <h1>
    Home
  </h1>
</div>
`;

exports[`test NavBar should render correctly in 1`] = `
<div
  className="navbar"
>
  <NavLink
    activeClassName="activeRoute"
    to="/settings"
  >
    <NavIcon
      iconurl="https://res.cloudinary.com/munnotubbel/image/upload/v1581240405/wombat/gear_fnbjbo.png"
      routename="Settings"
    />
  </NavLink>
  <NavLink
    activeClassName="activeRoute"
    to="/wallet"
  >
    <NavIcon
      iconurl="https://res.cloudinary.com/munnotubbel/image/upload/v1581240405/wombat/wallet_qg6uwn.png"
      routename="Wallet"
    />
  </NavLink>
  <NavLink
    activeClassName="activeRoute"
    to="/home"
  >
    <NavIcon
      iconurl="https://res.cloudinary.com/munnotubbel/image/upload/v1581240455/wombat/home_shuqs8.png"
      routename="Home"
    />
  </NavLink>
  <NavLink
    activeClassName="activeRoute"
    to="/explorer"
  >
    <NavIcon
      iconurl="https://res.cloudinary.com/munnotubbel/image/upload/v1581240405/wombat/compass_wzpihn.png"
      routename="Explorer"
    />
  </NavLink>
</div>
`;

exports[`test NavIcon should render correctly in 1`] = `
<div
  className="navicon"
>
  <img
    alt="NavButton"
  />
  <p />
</div>
`;

exports[`test ProgressBar should render correctly in 1`] = `
<div
  className="progressbar"
>
  <div
    className="percent"
  >
    <div>
      -
       %
    </div>
  </div>
  <div
    className="barOuter"
  >
    <div
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": "2px",
          "height": "16px",
          "maxWidth": "100%",
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;

exports[`test RessourceType should render correctly in 1`] = `
<div
  className="ressource_type"
>
  <div
    id="type"
  />
  <div
    id="consumption"
  />
  <div
    className="stakedinfo"
  >
    <p />
  </div>
</div>
`;

exports[`test Ressources should render correctly in 1`] = `
<div
  className="ressources"
>
  <div>
    <RessourceType
      consumption="0kb / 0kb"
      staked="0.0000"
      type="Net"
      withIcon={true}
    />
    <ProgressBar
      percent={0}
      slug="netBar"
    />
  </div>
  <div>
    <RessourceType
      consumption="0kb / 0kb"
      staked="0.0000"
      type="Cpu"
      withIcon={true}
    />
    <ProgressBar
      percent={0}
      slug="cpuBar"
    />
  </div>
  <div
    style={
      Object {
        "marginBottom": "60px",
      }
    }
  >
    <RessourceType
      consumption="0kb / 0kb"
      staked=""
      type="Ram"
    />
    <ProgressBar
      percent={0}
      slug="ramBar"
    />
  </div>
</div>
`;

exports[`test Settings should render correctly in 1`] = `
<div
  className="tobeadded"
>
  <h1>
    Settings
  </h1>
</div>
`;

exports[`test TradeButtons should render correctly in 1`] = `
<div
  className="tradebuttons"
>
  <div>
    <button
      className="tradebutton"
    >
      buy
    </button>
  </div>
  <div>
    <button
      className="tradebutton"
    >
      send
    </button>
  </div>
  <div>
    <button
      className="tradebutton"
    >
      sell
    </button>
  </div>
</div>
`;

exports[`test Wallet should render correctly in 1`] = `
<div
  className="wallet"
>
  <div
    className="ammount"
  >
    <EosLoader />
  </div>
  <TradeButtons />
  <div
    className="resourceinfo"
  >
    <div
      id="ressources_title"
    >
      <h4>
        Ressources
      </h4>
      <div>
        <h5>
          Staked
        </h5>
        <div
          className="stakedinfo"
        >
          <p>
            0.0000
          </p>
          <img
            alt="EOS-ICON"
            className="coinIcon"
            src="https://res.cloudinary.com/munnotubbel/image/upload/v1581182752/wombat/5a52232c2f93c7a8d5137fdd_izhtxc.png"
          />
        </div>
      </div>
    </div>
    <Ressources />
  </div>
</div>
`;
